python/src/ecole/core/observation.cpp

```diff
@@ -111,6 +112,33 @@ void bind_submodule(py::module m) {
 	def_reset(
 		node_bipartite, "Cache some feature not expected to change during an episode.");
 	def_obtain_observation(node_bipartite, "Extract a new :py:class:`NodeBipartiteObs`.");
+
+	py::class_<StrongBranchingScoresObs>(m, "StrongBranchingScoresObs", R"(
+		Strong Branching Scores for each variable at a branch-and-bound node.
+
+		This observation obtains scores for all pseudo-candidate variables at a branch-and-bound
```

Comment:
Hmm, maybe we should make that an option (pseudo-candidates or LP candidates), in the class constructor ?

Reply:
The only advantage I can see of doing that is to save a bit of time in strong branching if you know you will only branch on regular LP candidates (why compute the scores of the pseudo-candidates). But since you're already doing strong branching anyways, I'm not sure why you would care so much about time anyways. 

Instead, my thinking was, the observation would compute all possible scores, set the rest to np.nan, and let the ActionSet determine on which subset to focus.  Also, maybe you set the ActionSet to pseudo-branch candidates but you mistakenly set the Strong Branching to only compute LP candidates (setting the others to np.nan). Having all the possible scores computed by default solves this issue.

So at minimum, if we decide to add an argument in the constructor, in my opinion we should at least have the largest set possible as the default (that would be the pseudo-candidates, I guess). That would reduce potential mistakes for the end-user.

Reply:
There can be a pretty significant difference between computing psuedo-candidates and LP candidates so we may want to follow Maxime's suggestion.  I checked with an instance from MIPLIB and the difference was 45.98s with psuedo-candidates and 24.10s without for strong branching at the root node.  

Regarding the implementation of this, I think keeping the default to pseudo-candidates as Dider suggested would be a good idea.  It may also be possible to consider computing strong branching scores only for variables which are in the action set (i.e. only compute pseudo-candidates scores if the user would be branching on them).  This would take away the duplication of responsibility, but would not allow a user to want to branch on LP candidates and compute scores for pseudo-candidates.  Is it ever the case that someone would want to have access to the pseudo-candidate scores but would only be branching on LP candidates?

Reply:
>  It may also be possible to consider computing strong branching scores only for variables which are in the action set [...]
>  Is it ever the case that someone would want to have access to the pseudo-candidate scores but would only be branching on LP candidates?

I don't know but can the StrongBranchingObservation function even _know_ which ActionSet function was chosen?

Reply:
> > It may also be possible to consider computing strong branching scores only for variables which are in the action set [...]
> > Is it ever the case that someone would want to have access to the pseudo-candidate scores but would only be branching on LP candidates?
> 
> I don't know but can the StrongBranchingObservation function even _know_ which ActionSet function was chosen?

I am not sure, I would have to look into it further if we want to consider this as an option.

Reply:
It is now implemented as Dider suggested.